falcon/media_handlers.py

```diff
@@ -0,0 +1,101 @@
+import mimeparse
+from six.moves import UserDict
+
+from falcon import errors
+
+
+class Handlers(UserDict):
+    """Manages Media Type Handlers.
+
+    Attempts to load any imports for handlers as they are added.
+    """
+    def __init__(self, initial=None):
+        handlers = initial or {'application/json; charset=UTF-8': Json}
+
+        # Directly calling UserDict as it's not inheritable.
+        UserDict.__init__(self, handlers)
+
+    def update(self, input_dict):
+        for handler in input_dict.values():
+            handler.load()
+
+        UserDict.update(self, input_dict)
+
+    def __setitem__(self, key, item):
+        if not key:
+            raise ValueError('Media Type cannot be None or empty string')
+
+        item.load()
+        self.data[key] = item
+
+    def find_by_media_type(self, media_type, default):
```

Comment:
maybe we should define a default value and make default a keyword argument. In the usages below where find_by_media_type is called, we pass a default_value from the same object (self.options) from where we call media_handlers